src/evm/protocol/vm/erc20_token.rs

```diff
@@ -305,13 +309,28 @@ mod tests {
         assert_eq!(overwrites[&factory.token_address][&factory.total_supply_slot], supply);
     }
 
+    fn new_state() -> SimulationDB<RootProvider<BoxTransport>> {
+        let eth_rpc_url = env::var("ETH_RPC_URL").unwrap_or_else(|_| {
+            dotenv().expect("Missing .env file");
+            env::var("ETH_RPC_URL").expect("Missing ETH_RPC_URL in .env file")
```

Comment:
This is kinda weird and doesn't it check for the envvar twice?
```
    dotenv().ok();  // first we should load dotenv and return ok() because it's fine if .env doesn't exist in tests IMO
    
    // Then check for env var once
    let eth_rpc_url = env::var("ETH_RPC_URL")
        .expect("Missing ETH_RPC_URL in environment");
```